src/main/java/org/example/data/User.java

```diff
@@ -0,0 +1,37 @@
+package org.example.data;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+public class User {
+    private String userName;
+    private LocalDate regDate;
+    private List<Pony> ponyList = new ArrayList<>();
```

Comment:
if you dont want to save the same pony instance twice for the user you could use Set here